pages/cinema/[id].js

```diff
@@ -7,13 +7,27 @@ import PushButton from "../../components/PushButton";
 import Navigation from "../../components/Navigation";
 
 export default function MovieDetailPage() {
-  const { currentlyInCinemas } = useContext(CinemaContext);
+  const { currentlyInCinemas, upcomingMovies } = useContext(CinemaContext);
 
   const router = useRouter();
 
-  const currentMovie = currentlyInCinemas.find(
-    (movie) => movie.id.toString() === router.query.id
-  );
+  function getCinemaMovie() {
+    if (
+      currentlyInCinemas.find(
+        (movie) => movie.id.toString() === router.query.id
+      )
+    ) {
+      return currentlyInCinemas.find(
+        (movie) => movie.id.toString() === router.query.id
+      );
+    } else {
```

Comment:
You don't have to perform a `find` on `currentlyInCinemas` twice: store the result of the first `find` in a variable and return it if it's not undefined:
```js
const inCinemaMovie=currentlyInCinema.find((movie)=>movie.id===router.query.id);
return inCinemaMovie || upcomingMovies.find((movie)=>movie.id===router.query.id);
```

Reply:
Done in the next push. Thanks!